google/resumable_media/requests/download.py

```diff
@@ -87,37 +94,53 @@ def _write_to_stream(self, response):
             ~google.resumable_media.common.DataCorruption: If the download's
                 checksum doesn't agree with server-computed checksum.
         """
-        expected_md5_hash = _get_expected_md5(
-            response, self._get_headers, self.media_url
-        )
+        if self.checksum not in ["md5", "crc32c", None]:
+            raise ValueError("checksum must be ``'md5'``, ``'crc32c'`` or ``None``")
+
+        if self.checksum:
+            # `_get_expected_checksum()` may return None, in which case it will
+            # emit the info log `_MISSING_CHECKSUM`.
+            expected_checksum = _get_expected_checksum(
```

Comment:
Missed a potential refactor of code. This is a suggestion based on conditions being checked.

Could you make `_get_expected_checksum()` return checksum_object as well?

IIUC there's a check on checksum_type so you could remove the block below:

```python
        if expected_checksum is not None and self.checksum == "md5":
            checksum_object = hashlib.md5()
        elif expected_checksum is not None and self.checksum == "crc32c":
            checksum_object = _helpers._get_crc32c_object()
        else:
            md5_hash = hashlib.md5()
            checksum_object = _DoNothingHash()
```